lib/puppet/network/format_support.rb

```diff
@@ -35,7 +35,9 @@ def support_format?(name)
     end
 
     def supported_formats
-      result = format_handler.formats.collect do |f|
+      result = format_handler.formats.reject do |f|
+        f == :binary
```

Comment:
It feels a little odd to me to do this here. I'm still trying to trace through the implications but just to put it out there:

What if instead we removed this:
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/formats.rb#L82-L84

And specified `:required_formats => :render_method, :intern_method` for the `:binary` format?

This is from Luke, circa 2008 3101ea2:

>  Adding a hackish raw format.
    As the comment in the file says, we don't really have enough
    data to know what a good design would look like, and I think
    this format will be a bit of a one-off, so I'm just throwing
    up some barriers to keep people from doing silly things with it.

Then it's still in the generic list of `supported_formats` but won't be available for `Puppet::Resource::Catalog` etc